### DISCUSSION TITLE
Lesson7. testFundUpdatesFundedDataStructure failed.

### DISCUSSION THREAD

**Discussion by mibiot** [Cyfrin/foundry-full-course-cu#447] [Q&A] [answered]
Hi fellas,

New day new issue) At 1-17-11 Patric runs the testFundUpdatesFundedDataStructure test and it is passed.
Here is the screenshot of his piece of code:

<img width="835" alt="Screenshot 2023-07-25 at 19 56 12" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/55ad8ee5-e92c-4034-a4ea-36601afce520">

here is the code I m writing:
<img width="636" alt="Screenshot 2023-07-25 at 20 01 30" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/67d5aa03-e153-4907-95b1-921f98660a2a">

looks exactly the same. When I run this test I m getting an error:
<img width="1342" alt="Screenshot 2023-07-25 at 19 58 10" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/fce885cc-d202-488b-8540-d38e811a3f66">

just do not see any difference in the code so do not understand why it doesnt work for me. Looks like "vm.prank(USER);" works in the video.

Please advice. Thx

**Comment by ChefAharoni** [comment]
How does your script file looks like?

**Reply by mibiot** [reply]
<img width="631" alt="Screenshot 2023-07-26 at 12 22 55" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/140085328/4d145099-b2fa-40b8-a339-848e5e5a24f9">

**Comment by PatrickAlphaC** [comment]
Hmm... what happens when you git clone my code and try running the code I wrote?

**Reply by mibiot** [reply]
Hi Patric, 

First of all I would like to thank you for this awesome course and the opportunity to jump into this blockchain rabbithole absolutely free of charge just keeping interest and motivation in focus. You are doing a great job providing people with the opportunity to become a part of this fascinating digital space that has been changing the world. Lots of respect to you for this.

Concerning the topic theme - yes it has definitely started working when I replaced my files with the files from the repo. But the reason I posted the question - I was trying to figure out what was the reason my code didn’t work. While writing all of the above suddenly the answer has come from @Neftyr (thanks a lot for helping bro), - it is just a simple syntaxes error cause of fatigue that lead to another hours searching for the mistake. But these hours are not waisted, it helped me repeat the material and better understand some concepts.

Thanks again for the job you are making!

**Comment by Niferu** [comment, marked as answer]
Hello @mibiot 

Issue is in below line in your deploy script:

```solidity
vm.stopBroadcast;
```

It should be:
```solidity
vm.stopBroadcast();
```

**Reply by mibiot** [reply]
Hi Neftyr, thanks for your eagle-eye view), I didn’t see the issue, while had been looking through files over and over.

**Reply by Niferu** [reply]
I'm glad I could help! :)

**Comment by itznishant** [comment]
> Hello @mibiot
> 
> Issue is in below line in your deploy script:
> 
> ```solidity
> vm.stopBroadcast;
> ```
> 
> It should be:
> 
> ```solidity
> vm.stopBroadcast();
> ```

Based on this issue, I have got a doubt on the error displayed. 
I would like to understand why the solidity compiler not throw a compilation error in ```DeployFundMe.s.sol``` file as this looks like a syntax issue?

Could this be because any code after the ```vm``` keyword in  ```DeployFundMe.s.sol``` is being checked at run time and not while compiling.

Could someone clarify?
